hpx/traits/brace_initializable_traits.hpp

```diff
@@ -85,7 +85,8 @@ namespace hpx { namespace traits
 
     template <typename T, std::size_t N>
     constexpr auto is_paren_constructible()
-    -> detail::is_paren_constructible<T, std::make_index_sequence<N>>
+    -> decltype(detail::is_paren_constructible(
+            static_cast<T*>(nullptr), std::make_index_sequence<N>{}))
```

Comment:
Something's not right with this: http://cdash.cscs.ch//viewBuildError.php?buildid=45979 (those failures are coming from a test using this functionality). Unlike `is_brace_constructible`, `is_paren_constructible` doesn't have a two-argument constructor so this isn't going to work. I wouldn't dare to say if one should be added though. @wteiwewte  would you be able to have a look at this?

@m-diers if you have any more details about the failure that would probably help in fixing it.

Reply:
i suspect i did some crap yesterday. i can't reproduce it today and the test fails on me today, too.
i will close this pr first, and as soon as i have brought light into the dark again, i will get back to you.
sorry for the effort. ;-(

Reply:
No problem, thanks for looking into it!

Reply:
Found origin, it was an old gcc-toolchain which was used for some reason.